holidayextras/plugin-harvest
# Web 10870

#### What does this PR do? (please provide any background)

https://hxshortbreaks.atlassian.net/browse/WEB-10870

when updating a basket if the basket id doesn't exist in harvset currently the-works fails, this changes it to create a new basket when the put fails, with the same id that was passed in on the call to the put,

updates dependencies, 

#### What tests does this PR have?

fixed existing

#### How can this be tested?

- pull https://github.com/holidayextras/the-works/pull/517
- re-install your node_modules,
- start up the-works
- using [this](https://www.getpostman.com/collections/bcc4b167c0d42e328be2) postman collection
 - first call baskets create, then in the response grab the id from the JSON response and search for it in harvest on [cloudant](https://shortbreaks.cloudant.com/dashboard.html#database/harvest/_all_docs)
 - once found delete this doc
 - then open up the baskets update request replacing `0d4fc68ce0bf429d81b336f30361e18a ` with your id from the response of the create call, and call baskets update, doing this the-works should not error at all but return you a new basket with the same as the one you passed in


#### Any tech debt?

nope

#### Screenshots / Screencast

#### What gif best describes how you feel about this work?
![]()

- I have checked our general [contributing document](https://github.com/holidayextras/culture/blob/master/CONTRIBUTING.md) and the project specific [contributing document](../blob/master/CONTRIBUTING.md) (if present) and I'm happy for this to be reviewed.

By approving a review you are confirming you have...
- Witnessed the work behaving as expected (this could be on the author's machine or screencast).
- Checked for coding anti-patterns.
- Checked for appropriate test coverage.
- Checked all the tests are passing.

---
